src/traces/scatter/calc.js

```diff
@@ -115,6 +115,10 @@ module.exports = function calc(gd, trace) {
     for(i = 0; i < serieslen; i++) {
         cd[i] = (isNumeric(x[i]) && isNumeric(y[i])) ?
             {x: x[i], y: y[i]} : {x: false, y: false};
+
+        if(trace.identifier && trace.identifier[i] !== undefined) {
```

Comment:
> '7' and 7  and they'll be confused by the silent conversion

Good point. But, number of numeric strings are taking to be the same throughout the plotly.js API. Example: http://codepen.io/etpinard/pen/yJdJpj

So, this _shouldn't_ come as a surprise to plotly.js users.